eng/Subsets.props

```diff
@@ -453,6 +455,7 @@
   </ItemGroup>
 
   <ItemGroup Condition="$(_subset.Contains('+libs.ref+')) or $(_subset.Contains('+libs.src+')) or $(_subset.Contains('+libs.sfx+'))">
+    <ProjectToBuild Include="$(ToolsProjectRoot)illink\src\ILLink.Tasks\ILLink.Tasks.csproj" />
```

Comment:
Doing this means that we can't do a `dotnet build src\libraries\System.Text.Json\src\ -f net7.0` (any target framework, net7.0 is just an example) without first building via the build command script. That's because you use the Subsets system here to define dependencies, which only works for the root build.

Instead, we should add a ProjectReference to ILLink.Tasks.csproj to the projects that need it. Presumably that are all the projects that import illink.targets. Consider adding a ProjectReference to illink.targets and remove these entries here.

Reply:
> I was debating the best way to express the dependency. One problem is that ILLink.Tasks.csproj builds an MSBuild task, so I don't think building it as a ProjectReference would make the task available early enough for a UsingTask in the consuming project.

ProjectReferences are built as part of the `ResolveProjectReferences` target which runs before the CoreCompile target in which ILLink is invoked. The UsingTask only needs the path to the assembly at evaluation time, but the assembly doesn't need to exist at evaluation time.

The task subset is just another abomination which we eventually need to fix and declare dependencies in the consuming projects. I would recommend to ignore that.